drivers/include/periph/rtc_mem.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_periph_rtc_mem Low-Power RTC Memory
+ * @ingroup     drivers_periph_rtc
+ * @brief       Low-level RTC Memory peripheral driver
+ *
+ * This API provides an interface to access low-power memory present on some RTCs.
+ * This memory is retained even when the rest of the system is powered off.
+ *
+ * @{
+ * @file
+ * @brief       Low-level RTC memory peripheral driver interface definitions
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef PERIPH_RTC_MEM_H
+#define PERIPH_RTC_MEM_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
Do we need to tie this to `rtc`? For `stm32` this could be used regardless of the rtc.

Reply:
If STM32 have persistent GP registers across reset we can extend and move this interface somewhere later.